### ISSUE TITLE
[3.3.5][Spell][Hunter] Roar of Sacrifice - Critimmunity

### ISSUE THREAD

**Issue by Rushor** [TrinityCore/TrinityCore#15757]
rev: https://github.com/TrinityCore/TrinityCore/commit/b2575a04ce076aaaa0d0cb38737950b202127132
Since we are not allowed to post multiple issues in the same ticket, i will open this one here

related:
https://github.com/TrinityCore/TrinityCore/issues/13211
https://github.com/TrinityCore/TrinityCore/issues/7882

The first point in https://github.com/TrinityCore/TrinityCore/issues/7882 is present.

Issue: Roar of Sacrifice are not protecting the hunter 100% from crits high lvl crit champ still can crit them.

In other words:
If you get spell=53480 (roar of sacrifice) you should no longer get critical hits if you get hit by spell while the enemy is under the effect: spell=49796 (Deathchill)
It's strange that the critimmunity works with spells like lavaeruption + Flame Shock, but it does not work for spells like Deathchill

**Comment by Rushor** [comment]
i cannot reproduce this anymore...
The isspellcrit() check in unit.cpp seems to to it's job
